lib/clib/Corres_UL_C.thy

```diff
@@ -820,6 +820,37 @@ lemma ccorres_call:
   apply simp
   done
 
+lemma ccorres_call_getter_setter:
+  assumes cul: "ccorresG sr \<Gamma> (=) xf' P (i ` P') [] getter (Call f)"
+  and     gsr: "\<And>x x' s t rv rv'.
+                  \<lbrakk> (x, t) \<in> sr; rv = xf' t; (rv', x') \<in> fst (setter rv x) \<rbrakk>
+                  \<Longrightarrow> (x', g s t (clean s t)) \<in> sr"
+  and     res: "\<And>s t rv. rv = xf' t \<Longrightarrow> rv = xf (g s t (clean s t))"
+  and     ist: "\<And>x s. (x, s) \<in> sr \<Longrightarrow> (x, i s) \<in> sr"
+  and      ef: "\<And>val. empty_fail (setter val)"
+  shows "ccorresG sr \<Gamma> (=) xf P P' hs
+           (do val \<leftarrow> getter; setter val; return val od)
```

Comment:
thank you for calling these getter and setter as opposed to f and g

Reply:
Although it does suggest that `getter` doesn't change the state, which in our application it does. I'm happy with leaving it as getter/setter, though.